src/NuGet.Core/NuGet.Common/NuGetEnvironment.cs

```diff
@@ -193,6 +196,57 @@ private static string GetHome()
             }
         }
 
+        public static string GetDotNetLocation()
+        {
+            string path = Environment.GetEnvironmentVariable("PATH");
+            foreach (var dir in path.Split(':', ';'))
```

Comment:
Why split on colon? Wouldn't that break c:\foo ?
Also, do you need to stop off quotes and spaces from each PATH? Have to see what the OS does.